setup.py

```diff
@@ -103,7 +103,7 @@ def build_extension(self, ext):
     cmdclass={"build_ext": CMakeBuild},
     zip_safe=False,
     packages=find_namespace_packages(include=["mqt.*"]),
-    install_requires=["qiskit-terra>=0.20.2,<0.22.0"],
+    install_requires=["qiskit-terra>=0.20.2,<0.22.0", "retworkx>=0.11.0", "networkx>=1.9"],
```

Comment:
I don't really have experience with managing dependency versions. How do I decide on a maximum version? Simply the most recent one?

Reply:
Typically you choose the next higher version bump that is not yet available. For example, in case of `qiskit-terra`, the current release is `0.21.2`; so the upper bound is chosen as `<0.22.0`. In that fashion, future patch releases (which should not contain breaking changes) can be used automatically and dependabot will create a PR as soon as `0.22.x` drops in order to raise the upper bound.
This allows to maintain a range of supported versions for the dependencies and frequently avoids situations where two packages that depend on each other pin different versions of a dependency (which causes the installation to fail).